migrations_sqlite/2018-01-14-030655_switch_to_wal_mode/up.sql

```diff
@@ -0,0 +1,15 @@
+-- diesel_migrations runs each migration inside a transaction, and
+-- there is currently no way to opt-out of this behavior.  For most
+-- data definition tasks, this is fine.
+--
+-- However, the journal mode for an SQLite database cannot be set
+-- within a transaction.  To get around this, we close the Diesel
+-- transaction with COMMIT, set the journal mode, and then open
+-- another transaction that Diesel can finish.  It's kinda gross
+-- but ¯\_(ツ)_/¯
+
+COMMIT;
+
+PRAGMA journal_mode=WAL;
```

Comment:
This actually blocks the SQLite backend from running against an in-memory database, because WAL journal mode can't be used with in-memory databases (an in-memory database has its own journal mode).

This'll have to be set in the connection setup phase somehow.